File: py-sdk/src/forta_bot/utils/logger.py

class Logger:
    def __init__(self, is_prod: bool, is_debug: bool):
        self.is_prod = is_prod
        self.is_debug = is_debug

    def debug(self, msg):
        if not self.is_debug:
            return
        print(msg)

    def log(self, msg):
        if self.is_prod and not self.is_debug:
            return
        print(msg)

    def info(self, msg):
        print(msg)

    def error(self, msg):
        print(msg)

File: ts-sdk/src/utils/logger.ts
export class Logger {
  constructor(
    private readonly isProd: boolean,
    private readonly isDebug: boolean
  ) {}

  debug(message: any) {
    if (!this.isDebug) return;
    console.log(message);
  }

  log(message: any) {
    if (this.isProd && !this.isDebug) return;
    console.log(message);
  }

  info(message: any) {
    console.log(message);
  }

  error(message: any) {
    console.error(message);
  }
}
